go/saltpack/encrypt.go

```diff
@@ -30,11 +31,6 @@ type encryptStream struct {
 
 func (es *encryptStream) Write(plaintext []byte) (int, error) {
 
-	if !es.didHeader {
-		es.didHeader = true
-		es.err = es.encoder.Encode(es.header)
-	}
-
```

Comment:
`didHeader` is gone, and we now write out the header as part of `init`. (We also encode it to bytes now, rather than directly to the output stream, so that we can hash it.)